# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: widget/windows/TSFTextStore.cpp
# Commit: 60724af309c9
# Full Hash: 60724af309c94cfa3161b4533b00f11b9b5c73c0
# Author: Masayuki Nakano <masayuki@d-toybox.com>
# Date: 2022-02-08 17:32:27
# Regressor Bug: 1746104
# File Overlap Count: 1
# Description:
#   Bug 1746104 - part 1-5: Make `TSFTextStore::Selection` return const reference of `WritingMode` rather than a copy r=m_kato
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D137408
# ==============================================================================

diff -r 1dc1117af560 -r 60724af309c9 widget/windows/TSFTextStore.cpp
--- a/widget/windows/TSFTextStore.cpp	Mon Feb 07 22:33:32 2022 +0000
+++ b/widget/windows/TSFTextStore.cpp	Mon Feb 07 22:33:32 2022 +0000
@@ -4106,7 +4106,7 @@
           paAttrVals[count].varValue.vt = VT_BOOL;
           paAttrVals[count].varValue.boolVal =
               selectionForTSF.isSome() &&
-                      selectionForTSF->GetWritingMode().IsVertical()
+                      selectionForTSF->WritingModeRef().IsVertical()
                   ? VARIANT_TRUE
                   : VARIANT_FALSE;
           break;
@@ -4116,7 +4116,7 @@
           paAttrVals[count].varValue.vt = VT_I4;
           paAttrVals[count].varValue.lVal =
               selectionForTSF.isSome() &&
-                      selectionForTSF->GetWritingMode().IsVertical()
+                      selectionForTSF->WritingModeRef().IsVertical()
                   ? 2700
                   : 0;
           break;
@@ -7204,7 +7204,7 @@
     // XXX Do we need to set a new writing-mode here when setting a new
     // selection? Currently, we just preserve the existing value.
     WritingMode writingMode =
-        mSelection.isNothing() ? WritingMode() : mSelection->GetWritingMode();
+        mSelection.isNothing() ? WritingMode() : mSelection->WritingModeRef();
     mSelection = Some(TSFTextStore::Selection(mComposition->StartOffset(),
                                               mComposition->Length(), false,
                                               writingMode));